napalm_onyx/onyx_ssh.py

```diff
@@ -316,32 +316,33 @@ def _create_sot_file(self):
         self._send_config_commands(commands)
 
     def _get_diff(self):
-        """Get a diff between running config and a proposed file."""
+        """Get a diff between a proposed file and running config.
+
+        we don't have direct command to compare text file with running config
+        so will loop through file lines and comapre
+        """
         diff = []
-        self._create_sot_file()
-        command = ('show diff rollback-patch file {0} file {1}'.format(
-                   'sot_file', self.replace_file.split('/')[-1]))
-        diff_out = self.device.send_command(command)
-        try:
-            diff_out = diff_out.split(
-                'Generating Rollback Patch')[1].replace(
-                    'Rollback Patch is Empty', '').strip()
-            for line in diff_out.splitlines():
-                if line:
-                    if line[0].strip() != '!' and line[0].strip() != '.':
-                        diff.append(line.rstrip(' '))
-        except (AttributeError, KeyError):
-            raise ReplaceConfigException(
-                'Could not calculate diff. It\'s possible the given file doesn\'t exist.')
+        running_config = self.get_config(retrieve='running')['running']
+        running_lines = [line.strip() for line in running_config.splitlines()]
+
+        if self.replace_file is not None:  # this will be assigned when the file is uploaded successfully to switch
+            with open(self.replace_file, "rb") as f:
+                text = f.read()
```

Comment:
for line in f:

Reply:
we don't have unit test for these functions, just for show commands, do you have any idea how to start unit test for those?

I'll rename f